broker-node/app/DataMapping.php

```diff
@@ -22,90 +22,93 @@ public static function boot()
     protected $table = 'data_mappings';
 	protected $fillable = ['genesis_hash', 'hash', 'chunk_idx'];
     public $incrementing = false;  // UUID
-}
 
-/**
- * Builds and loads the datamap in DB given genesis hash and the number of file chunks
- *
- * @param string $genesis_hash A 64 char string
- * @param integer $file_chunk_count the number to generate
- * @return boolean $result response code
- */
-function buildMap($genesis_hash, $file_chunk_count) {
-    //we put the first chunk
-    $chunk_idx = 0;
-    App\DataMapping::create([
-        'genesis_hash' => $genesis_hash,
-        'hash' => $genesis_hash,
-        'chunk_idx' => $chunk_idx
-    ]);
-
-    //it takes the ceiling then disregards any past the correct number
-    $hashes_per_db_update = 5;
-    $num_groups = ceil($file_chunk_count / $hashes_per_db_update);
-
-    //start process of getting hashes
-    $hash_generator = hashGenerator($genesis_hash,  $file_chunk_count);
-
-    for($i = 0; $i < $num_groups; $i++) {
-        $next_group = getNextNHashes($hash_generator, $hashes_per_db_update, $chunk_idx);
-        $chunk_idx += $hashes_per_db_update;
-
-        foreach($next_group as $chunk_idx_hash) {
-            [$curr_chunk_idx, $curr_hash] = $chunk_idx_hash;
-
-            App\DataMapping::create([
-                'genesis_hash' => $genesis_hash,
-                'hash' => $curr_hash,
-                'chunk_idx' => $curr_chunk_idx
-            ]);
+    /**
+     * Builds and loads the datamap in DB given genesis hash and the number of file chunks
+     *
+     * @param string $genesis_hash A 64 char string
+     * @param integer $file_chunk_count the number to generate
+     * @return boolean $result response code
+     */
+    public static function buildMap($genesis_hash, $file_chunk_count) {
+        //we put the first chunk
+        $chunk_idx = 0;
+        self::create([
+            'genesis_hash' => $genesis_hash,
+            'hash' => $genesis_hash,
```

Comment:
shouldn't this be `'data' => $data`?

Reply:
ohhh you're saving the genesis hash as the first chunk

Reply:
We don't save the chunk to the DB, it gets saved on the tangle